pkg/klock/klock.go

```diff
@@ -344,11 +345,14 @@ func (p *Printer) updateColDefHeaders(objTable *metav1.Table, printNamespace boo
 			headers = append(headers, strings.ToUpper(colDef.Name))
 		}
 	}
+	for _, label := range labelCols {
+		headers = append(headers, strings.ToUpper(label))
```

Comment:
I noticed that `kubectl` seems to be only printing the part after the last slash in the label name.

For example: `kubectl get pods -L foo/bar/moo/doo` only shows a column named `DOO`. More importantly, a column like `app.kubernetes.io/version` has the header `VERSION`.

I wonder if you could just do a basic [`path.Base()`](https://pkg.go.dev/path#Base) (not the `filepath` variant). That should be sufficient

Reply:
Oh you know what, `path.Base()` might not be appropriate, as it has logic like `path.Base("") == "."`, whereas we want empty string input to equal empty string output.

Maybe just do a `strings.Split`, or maybe a `strings.LastIndexByte` if you're feeling adventurous

Reply:
Nice catch. I thought `path.Base()` would be enough too.

Reply:
Nice catch. I thought `path.Base()` would be enough.

As for your concern about the empty string, the cobra function `StringSliceVarP` does not consider empty strings as arguments.

Reply:
Well, can at least observe the same "ignore empty string" behavior on `kubectl`:

```console
$ kubectl get pods -L '' | bat -A
NAME··································READY···STATUS····RESTARTS········AGE␊
mastodon-postgresql-0·················1/1·····Running···0···············37h␊
mastodon-redis-master-0···············1/1·····Running···0···············37h␊
mastodon-sidekiq-79f7875dc6-245hb·····1/1·····Running···5·(4h47m·ago)···37h␊
mastodon-streaming-596898f5b5-9dk2r···1/1·····Running···0···············37h␊
mastodon-web-54f95b65db-flpzs·········1/1·····Running···0···············37h␊

$ kubectl get pods -L ',' | bat -A
NAME··································READY···STATUS····RESTARTS········AGE·········␊
mastodon-postgresql-0·················1/1·····Running···0···············37h·········␊
mastodon-redis-master-0···············1/1·····Running···0···············37h·········␊
mastodon-sidekiq-79f7875dc6-245hb·····1/1·····Running···5·(4h49m·ago)···37h·········␊
mastodon-streaming-596898f5b5-9dk2r···1/1·····Running···0···············37h·········␊
mastodon-web-54f95b65db-flpzs·········1/1·····Running···0···············37h·········␊
```

So that's fine.